File: Template_Floodfill_Dfs_733___1func.js
/**
 * @param {number[][]} image
 * @param {number} sr
 * @param {number} sc
 * @param {number} newColor
 * @return {number[][]}
 */
var floodFill = function(image, x, y, newColor) {
    let oldColor = image[x][y]
    if (oldColor == newColor)
        return image
    let c = image[0].length
    let r = image.length
    image[x][y] = newColor
    if (x < r - 1)
        floodFill(image, x + 1, y, newColor)
    if (x > 0)
        floodFill(image, x - 1, y, newColor)
    if (y < c - 1)
        floodFill(image, x, y + 1, newColor)
    if (y > 0)
        floodFill(image, x, y - 1, newColor)
    return image
};

File: Template_Floodfill_Dfs_733___1func.go
func floodFill(image [][]int, x int, y int, nc int) [][]int {
    if image[x][y] == nc {
        return image
    }
    oc := image[x][y]
    c := len(image[0])
    r := len(image)
    image[x][y] = nc
    if x + 1 < r && image[x + 1][y] == oc {
        floodFill(image, x + 1, y, nc)
    }
    if x - 1 > -1 && image[x - 1][y] == oc {
        floodFill(image, x - 1, y, nc)
    }
    if y + 1 < c && image[x][y + 1] == oc {
        floodFill(image, x, y + 1, nc)
    }
    if y - 1 > -1 && image[x][y - 1] == oc {
        floodFill(image, x, y - 1, nc)
    }
    return image
}
